FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1494 @@
+ * This demo uses a thread safe queue to hold commands for interacting with the
+ * MQTT API. A command task processes commands from the queue while other tasks
+ * enqueue them. In addition to the command task, this demo uses one task for
+ * publishing messages to the MQTT broker and another for receiving them via
+ * an MQTT subscription. Each task has a queue to hold received publish messages,
+ * and the command task pushes incoming publishes to the queue of each task
+ * that is subscribed to the incoming topic.
+ *
+ * !!! NOTE !!!
+ * This MQTT demo does not authenticate the server nor the client.
+ * Hence, this demo should not be used as production ready code.
+ */
+
+/* Standard includes. */
+#include <string.h>
+#include <stdio.h>
+#include <assert.h>
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "queue.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_Sockets.h"
+
+/* Demo Specific configs. */
+#include "demo_config.h"
+
+/* MQTT library includes. */
+#include "mqtt.h"
+
+/* Transport interface include. */
+#include "plaintext_freertos.h"
+
+/**
+ * These configuration settings are required to run the demo.
+ * Throw compilation error if the below configs are not defined.
+ */
+#ifndef CLIENT_IDENTIFIER
+    #error "Please define a unique CLIENT_IDENTIFIER."
+#endif
+#ifndef BROKER_ENDPOINT
+    #error "Please define BROKER_ENDPOINT."
+#endif
+
+/**
+ * Provide default values for undefined configuration settings.
+ */
+#ifndef BROKER_PORT
+    #define BROKER_PORT    ( 1883 )
+#endif
+
+#ifndef NETWORK_BUFFER_SIZE
+    #define NETWORK_BUFFER_SIZE    ( 1024U )
+#endif
+
+/**
+ * @brief Length of client identifier.
+ */
+#define CLIENT_IDENTIFIER_LENGTH            ( ( uint16_t ) ( sizeof( CLIENT_IDENTIFIER ) - 1 ) )
+
+/**
+ * @brief Length of MQTT server host name.
+ */
+#define BROKER_ENDPOINT_LENGTH              ( ( uint16_t ) ( sizeof( BROKER_ENDPOINT ) - 1 ) )
+
+/**
+ * @brief Timeout for receiving CONNACK packet in milliseconds.
+ */
+#define CONNACK_RECV_TIMEOUT_MS             ( 1000U )
+
+/**
+ * @brief Timeout for MQTT_ProcessLoop function in milliseconds.
+ */
+#define MQTT_PROCESS_LOOP_TIMEOUT_MS        ( 200U )
+
+/**
+ * @brief The maximum time interval in seconds which is allowed to elapse
+ *  between two Control Packets.
+ *
+ *  It is the responsibility of the Client to ensure that the interval between
+ *  Control Packets being sent does not exceed the this Keep Alive value. In the
+ *  absence of sending any other Control Packets, the Client MUST send a
+ *  PINGREQ Packet.
+ */
+#define MQTT_KEEP_ALIVE_INTERVAL_SECONDS    ( 60U )
+
+/**
+ * @brief Transport timeout in milliseconds for transport send and receive.
+ */
+#define TRANSPORT_SEND_RECV_TIMEOUT_MS      ( 20 )
+
+#define _MILLISECONDS_PER_SECOND            ( 1000U )                                                 /**< @brief Milliseconds per second. */
```

Comment:
why is the @brief comment different for this MACRO and the one below? I think it is better to be consistent with other MACRO comments.

Reply:
Probably since these macros are copy pasted from the plaintext demo. I'll change the macro name too, I don't like the leading underscore